src/main/java/com/redhat/devtools/intellij/tektoncd/listener/SaveInEditorListener.java

```diff
@@ -0,0 +1,96 @@
+
+import java.io.IOException;
+
+import static com.redhat.devtools.intellij.tektoncd.Constants.KIND_PLURAL;
+import static com.redhat.devtools.intellij.tektoncd.Constants.NOTIFICATION_ID;
+
+public class SaveInEditorListener extends FileDocumentSynchronizationVetoer {
+    Logger logger = LoggerFactory.getLogger(SaveInEditorListener.class);
+
+    @Override
+    public boolean maySaveDocument(@NotNull Document document, boolean isSaveExplicit) {
+        VirtualFile vf = FileDocumentManager.getInstance().getFile(document);
+        // if file is not related to tekton we can skip it
+        if (vf == null || vf.getUserData(KIND_PLURAL).isEmpty()) {
+            return true;
+        }
+
+        String namespace, name, apiVersion;
+        JsonNode spec;
+        CustomResourceDefinitionContext crdContext;
+        try {
+            namespace = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "namespace"});
+            name = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "name"});
+            if (Strings.isNullOrEmpty(namespace) || Strings.isNullOrEmpty(name)) {
+                throw new IOException("Tekton file has not a valid format. Namespace and/or name fields are invalid.");
+            }
+            apiVersion = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"apiVersion"});
+            if (Strings.isNullOrEmpty(apiVersion)) {
+                throw new IOException("Tekton file has not a valid format. ApiVersion field is not found.");
+            }
+            crdContext = CRDHelper.getCRDContext(apiVersion, vf.getUserData(KIND_PLURAL));
+            if (crdContext == null) {
+                throw new IOException("Tekton file has not a valid format. ApiVersion field contains an invalid value.");
+            }
+            spec = YAMLHelper.getValueFromYAML(document.getText(), new String[] {"spec"});
+            if (spec == null) {
+                throw new IOException("Tekton file has not a valid format. Spec field is not found.");
+            }
+        } catch (IOException e) {
+            UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Error"));
+            return false;
+        }
+
+        Notification notification;
+        try (final KubernetesClient client = new DefaultKubernetesClient()) {
+            JsonNode customResource = JSONHelper.MapToJSON(client.customResource(crdContext).get(namespace, name));
+            ((ObjectNode) customResource).set("spec", spec);
+            client.customResource(crdContext).edit(namespace, name, customResource.toString());
+        } catch (IOException e) {
+            // give a visual notification to user if an error occurs during saving
+            notification = new Notification(NOTIFICATION_ID, "Error", "An error occurred while saving " + vf.getUserData(KIND_PLURAL) + " " + name + "\n" + e.getLocalizedMessage(), NotificationType.ERROR);
```

Comment:
Be coherent with line 92, so also call StringUtils.capitalize

Reply:
Heck if you are picky :grin: